hack/validate-vendor.bash

```diff
@@ -8,26 +8,32 @@
 
 set -e -o pipefail
 
-# TRAVIS_BRANCH is the name of the branch targeted by the pull request.
-# but we won't have that set if running locally
-if [ -z "$TRAVIS_BRANCH" ]; then
-	VALIDATE_REPO='git@github.com:golang/dep.git'
+if [ -z "$VALIDATE_UPSTREAM" ]; then
+	VALIDATE_REPO='https://github.com/golang/dep.git'
 	VALIDATE_BRANCH='master'
 
+	VALIDATE_HEAD="$(git rev-parse --verify HEAD)"
+
 	git fetch -q "$VALIDATE_REPO" "refs/heads/$VALIDATE_BRANCH"
-	TRAVIS_BRANCH="$(git rev-parse --verify FETCH_HEAD)"
-fi
+	VALIDATE_UPSTREAM="$(git rev-parse --verify FETCH_HEAD)"
 
-VALIDATE_HEAD="$(git rev-parse --verify HEAD)"
+	VALIDATE_COMMIT_DIFF="$VALIDATE_UPSTREAM...$VALIDATE_HEAD"
+
+	validate_diff() {
+		if [ "$VALIDATE_UPSTREAM" != "$VALIDATE_HEAD" ]; then
+			git diff "$VALIDATE_COMMIT_DIFF" "$@"
+		fi
+	}
+fi
 
 IFS=$'\n'
-files=( $(git diff "$TRAVIS_BRANCH...$VALIDATE_HEAD" --diff-filter=ACMR --name-only -- 'manifest.json' 'lock.json' 'vendor/' || true) )
+files=( $(validate_diff --diff-filter=ACMR --name-only -- 'manifest.json' 'lock.json' 'vendor/' || true) )
```

Comment:
Since #342 was just merged, this should be updated to `manifest.json -> Gopkg.toml` and `lock.json -> Gopkg.lock`.